arviz/data/io_pyro.py

```diff
@@ -107,24 +136,38 @@ def log_likelihood_to_xarray(self):
                 return None
         return dict_to_dataset(data, library=self.pyro, coords=self.coords, dims=self.dims)
 
-    @requires("posterior_predictive")
-    def posterior_predictive_to_xarray(self):
-        """Convert posterior_predictive samples to xarray."""
+    def translate_posterior_predictive_dict_to_xarray(self, dct, dims):
+        """Convert posterior_predictive or prediction samples to xarray."""
         data = {}
-        for k, ary in self.posterior_predictive.items():
+        for k, ary in dct.items():
             ary = ary.detach().cpu().numpy()
             shape = ary.shape
             if shape[0] == self.nchains and shape[1] == self.ndraws:
                 data[k] = ary
             elif shape[0] == self.nchains * self.ndraws:
                 data[k] = ary.reshape((self.nchains, self.ndraws, *shape[1:]))
+            elif self.num_chains is not None:
+                draws = shape[0] // self.num_chains
+                data[k] = ary.reshape((self.num_chains, draws, *shape[1:]))
             else:
                 data[k] = utils.expand_dims(ary)
                 _log.warning(
                     "posterior predictive shape not compatible with number of chains and draws."
                     "This can mean that some draws or even whole chains are not represented."
                 )
-        return dict_to_dataset(data, library=self.pyro, coords=self.coords, dims=self.dims)
+        return dict_to_dataset(data, library=self.pyro, coords=self.coords, dims=dims)
+
+    @requires("posterior_predictive")
+    def posterior_predictive_to_xarray(self):
+        """Convert posterior_predictive samples to xarray."""
+        return self.translate_posterior_predictive_dict_to_xarray(
+            self.posterior_predictive, self.dims
+        )
+
+    @requires("predictions")
+    def predictions_to_xarray(self):
+        """Convert predictions (out of sample predictions) to xarray."""
```

Comment:
I would either remove the out of sample predictions or change it to out of sample posterior predictive